src/client/Client.cc

```diff
@@ -4784,6 +4784,9 @@ void Client::trim_caps(MetaSession *s, uint64_t max)
     // is deleted inside remove_cap
     ++p;
 
+    if (in->dirty_caps || in->cap_snaps.size())
+      cap_delay_requeue(in.get());
```

Comment:
I think also, if the Inode has to flush dirty caps then we should not include it it as "trimmed" (line 4827 below).

Reply:
Makes sense.

Reply:
Makes sense and I will fix it. Thanks @batrick @vshankar